### PR TITLE
Improve UI in course pages

### LINKED ISSUE #117: Improve UI in course related pages
**Requirements :**

- Add css and change html in course related pages to improve the UI

### PR DESCRIPTION
**Pull request by RanyWh**
**Imporving the UI in the course related pages.**

**Changes made :**

- Add photo to the add course page
- Add style to add course page
- Add style to course table page
- Connect teacher photo in the course page
- Fix bug where the course icon wasn't working in course page
- Show course table according to user (teacher/student/both)
- Add request to join course tab in course page
- Add column to show how many request to join there is in a course 
- Adjust tests to changes

Course table page :

image

Add course page :

צילום מסך 2023-06-01 204036

Course page:
image

fixes #117

### CODE REVIEW

File: course/templates/course/courses_two_tables.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            <tbody>
+                <td><a href="/course/{{course.course_id}}">{{course.course_name.capitalize}}</td>
+                <td>{{course.price}} ₪</td>
+                <td>{{course.get_category_display.capitalize}}</td>
+                <td>{{course.get_pending_requset_count_for_course}}</td>
+            </tbody>
+            {% endfor %}
+        </table>
+        <a href="add">
+            <!-- <button type="button" class="btn btn-success" style="float: right;">Add Course</button></a> -->
```
Comment (shirazyomtov) [inline course/templates/course/courses_two_tables.html:33]: Please remove this comment

File: static/styles/courses.table.css [hunk 2]
```diff
@@ -4,5 +4,116 @@ height: 50px;
...
+  left: 15px;
+  height: 4px;
+  top: calc(50% - 2px);
+}
+
+.add-btn:hover .add-icon::after {
+  right: 15px;
+  height: 4px;
+  top: calc(50% - 2px);
+}
```
Comment (shirazyomtov) [inline static/styles/courses.table.css:119]: Missing a new line at the end of the file

File: static/styles/student_course.css [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  padding-bottom: 10px;
+  left: 0.96em;
+  top: 0.7em;
+  width: 0.5em;
+  height: 0.75em;
+  border: solid white;
+  border-width: 0.2em 0.2em 0.2em 0.2em;
+  box-shadow: 0.1em 0.1em 0em 0 rgba(0,0,0,0.3);
+  transform: rotate(45deg);
+} */
```
Comment (shirazyomtov) [inline static/styles/student_course.css:48]: Please remove this comment

File: tests/course/test_course_pages.py [hunk 4]
```diff
@@ -20,33 +23,61 @@ def teacher_course_zero(persist_user):
...
+@pytest.fixture
+def authorized_client_teacher(client, persist_profile_teacher):
+    client.force_login(persist_profile_teacher.user)
+    return client
+
+
+@pytest.fixture
+def authorized_client_teacher_and_student(client, persist_profile_teacher_and_student):
+    client.force_login(persist_profile_teacher_and_student.user)
     return client
```
Comment (shirazyomtov) [inline tests/course/test_course_pages.py:50]: Maybe we could move the fixtures to the conftest.py

File: course/templates/course/courses_table.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    {% if user.profile.account_type == 'T'%}
+                    <td>{{course.get_pending_requset_count_for_course}}</td>
+                    {% endif %}
+                </tbody>
+                {% endfor %}
+            </table>
+        </div>
+        {% if user.profile.account_type == 'T'%}
+        <a href="add">
+            <!-- <button type="button" class="btn btn-success" style="float: right;">Add Course</button></a> -->
```
Comment (shirazyomtov) [inline course/templates/course/courses_table.html:45]: Please remove this comment

**Review by shirazyomtov — CHANGES_REQUESTED** [review]
Great work, left a few comments.

**Review by Davidco94 — APPROVED** [review]
LGTM

**Review by shirazyomtov — APPROVED** [review]
LGTM

**Comment by galitf** [comment]
I would split this MR according to the UI pages you are  improving: Add course page, course table page and course page.  Usually fixing a bug should be in a separate MR.

**Review by Omeramsc — APPROVED** [review]
This patch includes way too many changes and additions and can probably split to multiple PRs, as Galit suggested. But... as we are in the final week of the project, we better let it slide; do note that on your future works.

**Review by natifridman — APPROVED** [review]
LGTM

**Pull request merged**